xarray/tests/test_backends.py

```diff
@@ -1134,6 +1134,23 @@ def test_encoding_kwarg_compression(self):
 
         assert ds.x.encoding == {}
 
+    def test_keep_chunksizes_if_no_original_shape(self):
+        orig_ds = Dataset({'x': [1, 2, 3]})
+        chunksizes = (2, )
+        orig_ds.variables['x'].encoding = {
+            'chunksizes': chunksizes
+        }
+
+        with create_tmp_file() as tmp_file:
+            orig_ds.to_netcdf(path=tmp_file)
+
+            nc = nc4.Dataset(tmp_file, mode='r')
+            with backends.NetCDF4DataStore(nc, autoclose=False) as store:
+                with open_dataset(store) as ds:
```

Comment:
Looks like there's something wrong with the test. Can you make this just `with open_dataset(tmp_file) as ds` and avoid using `nc4.Dataset, backends.NetCDF4DataStore` etc.? 

You should be able to run the tests locally. See https://xarray.pydata.org/en/stable/contributing.html#running-the-test-suite for instructions

Reply:
It have fixed it as you suggested.

Sorry, it has been a while since I wrote this ;)